### PR TITLE
Add new instructions for VPC peering

### LINKED ISSUE #2497: Update AWS VPC peering creation doc based on new procedure
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://docs.astronomer.io/astro/connect-aws?tab=VPC%20peering#connection-options

### The Issue or Context

Process was reviewed and some steps were automated. All details are describe at [Notion page](https://www.notion.so/astronomerio/VPC-Peering-c521225ef87d4a13ade043d7b3dbf8a0).

### Fix or Proposed Change

Please reflect process from Notion in Doc page for public access.

### Additional Notes

_No response_

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by manmeetkaur**
Resolves #2497

### CODE REVIEW

File: astro/connect-aws.md [hunk 1]
```diff
@@ -48,15 +48,29 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
 
-To create a VPC peering connection between an Astro VPC and an AWS VPC, contact [Astronomer support](https://cloud.astronomer.io/support) and provide the following information:
+To create a VPC peering connection between an Astro VPC and an AWS VPC, complete the following steps:
 
-- Astro cluster ID and name
-- AWS Account ID of the target VPC
-- Region of the target VPC
-- VPC ID of the target VPC
-- Classless Inter-Domain Routing (CIDR) block of the target VPC
+1. In your AWS console, create a role using the [role creation stack template](https://docs.astronomer.io). In the template, enter a meaningful name for your stack. Then, enter your Astro AWS account ID as **Peer Owner IDs**. Other fields are optional. Follow [AWS documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-add-tags.html) to review and finish creating the stack.
```
Comment (manmeetkaur) [inline astro/connect-aws.md:53]: the VPC accepting AWS account. (their Data Stack not our Astro DP)

File: astro/connect-aws.md [hunk 2]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
 
 To set up a private connection between an Astro VPC and an AWS VPC, you can create a VPC peering connection. VPC peering ensures private and secure connectivity, reduces network transit costs, and simplifies network layouts.
 
-To create a VPC peering connection between an Astro VPC and an AWS VPC, contact [Astronomer support](https://cloud.astronomer.io/support) and provide the following information:
+To create a VPC peering connection between an Astro VPC and an AWS VPC, a temporary assumable role needs to be created. This role will be used by the Astro AWS account to initiate a VPC peering connection. Follow these steps to complete this process:
```
Comment (jwitz) [inline astro/connect-aws.md:51]:
```suggestion
To create a VPC peering connection between an Astro VPC and an AWS VPC, you must create a temporary assumable role. The Astro AWS account will assume this role to initiate a VPC peering connection.
```

File: astro/connect-aws.md [hunk 3]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
 
-- Astro cluster ID and name
-- AWS Account ID of the target VPC
-- Region of the target VPC
-- VPC ID of the target VPC
-- Classless Inter-Domain Routing (CIDR) block of the target VPC
+1. Go to the AWS console of your AWS account with the target VPC, and copy the following:
```
Comment (jwitz) [inline astro/connect-aws.md:53]:
```suggestion
1. Open the AWS console of the AWS account with the target VPC and copy the following:
```

File: astro/connect-aws.md [hunk 4]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+    - AWS account ID 
+    - AWS region
+    - VPC ID of the target VPC
+    - CIDR block of the target VPC
+    
+2. Create a temporary role using the [role creation stack template](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review?templateURL=https://cre-addon-infrastructure-us-east-1.s3.amazonaws.com/astro-peering-role.yaml). In the **Quick create stack** template that opens, enter the following info:
+
+    - A meaningful name for your stack in **Stack name**. 
+    - Your Astro AWS account ID as **Peer Owner IDs**.
```
Comment (jwitz) [inline astro/connect-aws.md:63]:
```suggestion
    - **Peer Owner IDs**: Enter your Astro AWS account ID.
```
Where do you find the account ID? Is it for Org or cluster?

File: astro/connect-aws.md [hunk 5]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+1. Go to the AWS console of your AWS account with the target VPC, and copy the following:
+
+    - AWS account ID 
+    - AWS region
+    - VPC ID of the target VPC
+    - CIDR block of the target VPC
+    
+2. Create a temporary role using the [role creation stack template](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review?templateURL=https://cre-addon-infrastructure-us-east-1.s3.amazonaws.com/astro-peering-role.yaml). In the **Quick create stack** template that opens, enter the following info:
+
+    - A meaningful name for your stack in **Stack name**.
```
Comment (jwitz) [inline astro/connect-aws.md:62]:
```suggestion
    - **Stack name**: Enter a meaningful name for your stack.
```

File: astro/connect-aws.md [hunk 6]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
-- VPC ID of the target VPC
-- Classless Inter-Domain Routing (CIDR) block of the target VPC
+1. Go to the AWS console of your AWS account with the target VPC, and copy the following:
+
+    - AWS account ID 
+    - AWS region
+    - VPC ID of the target VPC
+    - CIDR block of the target VPC
+    
+2. Create a temporary role using the [role creation stack template](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review?templateURL=https://cre-addon-infrastructure-us-east-1.s3.amazonaws.com/astro-peering-role.yaml). In the **Quick create stack** template that opens, enter the following info:
```
Comment (jwitz) [inline astro/connect-aws.md:60]:
```suggestion
2. Create a temporary role using the [role creation stack template](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review?templateURL=https://cre-addon-infrastructure-us-east-1.s3.amazonaws.com/astro-peering-role.yaml). In the **Quick create stack** template that opens, complete the following fields:
```

File: astro/connect-aws.md [hunk 7]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+    - A meaningful name for your stack in **Stack name**. 
+    - Your Astro AWS account ID as **Peer Owner IDs**.
+
+    Then, click on **Create stack**.
```
Comment (jwitz) [inline astro/connect-aws.md:65]:
```suggestion
```
This part is implied from our previous statement to "Create a stack"

File: astro/connect-aws.md [hunk 8]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+3. After the stack is created, go to the **Stack info** tab and copy the assume role arn from the **Stack ID** field.
```
Comment (jwitz) [inline astro/connect-aws.md:67]:
```suggestion
3. After the stack is created, go to the **Stack info** tab and copy the AssumeRole ARN from the **Stack ID** field.
```

File: astro/connect-aws.md [hunk 9]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+4. In your Cloud UI, click on the Astronomer logo in the top left corner. Then, go to the **Clusters** tab, select your cluster and copy the **ID** of your cluster.
```
Comment (jwitz) [inline astro/connect-aws.md:69]:
```suggestion
4. In the Cloud UI, click your Workspace name in the upper left corner, then click **Organization Settings**. Click **Clusters**, select your cluster, and copy the **ID** of the cluster.
```

File: astro/connect-aws.md [hunk 10]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+5. Contact [Astronomer support](https://cloud.astronomer.io/support) and provide the following details:
+
+    - **Stack ID** from step 3
+    - Astro cluster **ID** from step 4
+    - AWS region of the target VPC
+    - VPC ID of the target VPC
+    - AWS account ID of the target VPC
+    - CIDR block of the target VPC
```
Comment (jwitz) [inline astro/connect-aws.md:78]:
```suggestion
    - AWS region of the target VPC from Step 1
    - VPC ID of the target VPC from Step 1
    - AWS account ID of the target VPC from Step 1
    - CIDR block of the target VPC from Step 1
    - **Stack ID** from Step 3
    - Astro cluster **ID** from Step 4
```

File: astro/connect-aws.md [hunk 11]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+    
+    Astronomer support will initiate a peering request and create the routing table entries in the Astro VPC.
+
+6. Wait for the Astronomer support to notify you with the Astro VPC CIDR. Then, the owner of the target VPC needs to [accept the peering request](https://docs.aws.amazon.com/vpc/latest/peering/accept-vpc-peering-connection.html) and [create the routing table entries](https://docs.aws.amazon.com/vpc/latest/userguide/WorkWithRouteTables.html#AddRemoveRoutes) in the target VPC.
```
Comment (jwitz) [inline astro/connect-aws.md:82]:
```suggestion
6. Wait for Astronomer support to send you the Astro VPC CIDR. Then, the owner of the target VPC needs to [accept the peering request](https://docs.aws.amazon.com/vpc/latest/peering/accept-vpc-peering-connection.html) and [create the routing table entries](https://docs.aws.amazon.com/vpc/latest/userguide/WorkWithRouteTables.html#AddRemoveRoutes) in the target VPC.
```

File: astro/connect-aws.md [hunk 12]
```diff
@@ -48,15 +48,41 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+7. (Optional) You can now delete the stack that was created for role creation. This will delete the temporary assumable role.
```
Comment (jwitz) [inline astro/connect-aws.md:84]:
```suggestion
7. (Optional) Delete the stack that you created. This will delete the temporary assumable role.
```

**Review by jwitz — CHANGES_REQUESTED** [review]
One more Q about Astro account ID usage, besides that LGTM

File: astro/connect-aws.md (hunk 4 above)
Comment (manmeetkaur) [reply astro/connect-aws.md:63]: will add the same note as in the other PR: "To retrieve your cluster's AWS account ID on Astro Hosted, [Astronomer support](https://cloud.astronomer.io/support). To retrieve your cluster's AWS account ID in the Astro Hybrid, click the name of your Workspace in the upper left corner of the Cloud UI, click **Organization Settings**, then click **Clusters**. Open your cluster and copy its **Account ID**."

**Comment by manmeetkaur** [comment]
@jwitz ready for review.

File: astro/connect-aws.md [hunk 13]
```diff
@@ -48,15 +48,39 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+    - AWS account ID 
+    - AWS region
+    - VPC ID of the target VPC
+    - CIDR block of the target VPC
+    
+2. Create a temporary role using the [role creation stack template](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review?templateURL=https://cre-addon-infrastructure-us-east-1.s3.amazonaws.com/astro-peering-role.yaml). In the **Quick create stack** template that opens, complete the following fields:
+
+    - **Stack name**: Enter a meaningful name for your stack.
+    - **Peer Owner IDs**: Enter your Astro AWS account ID.
```
Comment (jwitz) [inline astro/connect-aws.md:63]:
```suggestion
    - **Peer Owner IDs**: Enter your cluster's AWS account ID. To retrieve your cluster's AWS account ID on Astro Hosted, contact [Astronomer support](https://cloud.astronomer.io/support). To retrieve your cluster's AWS account ID on Astro Hybrid, click the name of your Workspace in the upper left corner of the Cloud UI, click **Organization Settings**, then click **Clusters**. Open your cluster and copy its **Account ID**.
```

**Review by jwitz — APPROVED** [review]

**Pull request merged**